### PR TITLE
Origin Installation Documentation

### PR DESCRIPTION
**Pull request by turetske**
First draft of origin installation documentation. Please feel free to comment.

### CODE REVIEW

**Review by CannonLock — CHANGES_REQUESTED** [review]
Couple items to align with standard practice: 

- Merge with main and move this to /docs/pages/.
- Make the extension .mdx ( It isn't necessary but consistency is nice )
- Remove the top yaml
  - The docs framework will grab the file update time by default 
  - The title is presumed from file name, but you can also adjust with _meta.json file as specified in the linked docs of the README
  
This is a personal preference so no need to change but # are far superior for creating headers then the line ['---', '==='] notation. 

https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers

File: docs/origin_install.md
Comment (joereuss12) [inline docs/origin_install.md:88]: You have this example twice here.

File: docs/origin_install.md
Comment (joereuss12) [inline docs/origin_install.md:105]: I don't know much about markdown so I could be wrong, but maybe reformat these warnings to look more visually appealing. Could just be github not recognizing '!!!' or something too.

**Review by joereuss12 — CHANGES_REQUESTED** [review]
Just have a few suggestions but otherwise I think it looks pretty good. I haven't looked into installing an origin too much but could follow this along quite well. I feel everything should make sense especially for someone who would be an admin.

File: docs/origin_install.md
Comment (jhiemstrawisc) [inline docs/origin_install.md:41]: Is there actually any requirement that an admin configure XRootD? Looking through the xrootd-related steps below, I think most of this stuff is specifically stuff we auto-configure via the origin.

File: docs/origin_install.md
Comment (jhiemstrawisc) [inline docs/origin_install.md:125]: We should probably be consistent with what we call this step in our CLI and say "Serving an Origin"

File: docs/origin_install.md
Comment (jhiemstrawisc) [inline docs/origin_install.md:130]: This is missing the `serve` sub-command, and the fact that the argument to the volume flag takes two pieces. We need ```pelican origin serve -f <federation url> -v <local directory>:<namespace prefix>```

File: docs/origin_install.md
Comment (jhiemstrawisc) [inline docs/origin_install.md:132]: `<local directory>` -- the directory containing objects to be exported to the federation
`<namespace prefix>` -- the namespace at which files from `<local directory>` will be made available in the federation. For example, if I want to export `/test/foo.txt` to the federation, passing `-v /test:/my/namespace` will allow others to get `foo.txt` by doing an object copy against the path `/my/namespace/foo.txt`.

File: docs/origin_install.md
Comment (jhiemstrawisc) [inline docs/origin_install.md:154]: I'm not sure what we should say about registering the origin with the namespace registry. Eventually we want this to happen automatically. For now, the reality is that after serving the origin (which generates the asymmetric key pair needed by the registry) we also need to `pelican namespace register --prefix <namespace prefix> --namespace-url <registry url>`. This gives the registry the public key for the origin, and makes that public key available to all of the other services in the federation for verifying tokens.

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

File: docs/origin_install.md
Comment (turetske) [reply docs/origin_install.md:105]: Hmm. I can't see a good way to do this. At least in the markdown nextra uses

**Comment by turetske** [comment]
Addressed the changes, unfortunately, these instructions will all probably change with Brian's latest PR, but it's correct as of now.

File: docs/pages/origin_install.mdx
Comment (jhiemstrawisc) [inline docs/pages/origin_install.mdx:56]: The image and other console examples all show Pelican being run as root - is this something we require, or is it just an artifact from our developing in a rootly container?

File: docs/pages/origin_install.mdx
Comment (jhiemstrawisc) [inline docs/pages/origin_install.mdx:70]: This step should technically occur before serving the origin so that the health tests querying the director work. Can you move it up and explain what this step means?

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

**Comment by jhiemstrawisc** [comment]
One other quick comment -- it looks like each tab in the rendered index just reads the file name. Can you name the document something like `serving_an_origin.mdx`?

**Review by jhiemstrawisc — APPROVED** [review]
LGTM

**Comment by jhiemstrawisc** [comment]
@joereuss12 @CannonLock, when you get a second can you confirm that Emma made your requested changes and approve the PR? I can't merge until we've all signed off.

**Review by joereuss12 — APPROVED** [review]
LGTM!

**Review by CannonLock — APPROVED** [review]
LGTM

**Pull request merged**
